control/grpc.py

```diff
@@ -236,41 +265,177 @@ def __init__(self, config: GatewayConfig, gateway_state: GatewayStateHandler, rp
         self.subsys_max_ns = {}
         self.host_info = SubsystemHostAuth()
 
-    def create_host_psk_file(self, subsysnqn : str, hostnqn : str, psk_value : str) -> str:
+    def get_directories_for_key_file(self, key_type : str, subsysnqn : str, create_dir : bool = False) -> []:
+        tmp_files = []
+        try:
+            tmp_files = os.listdir("/tmp")
```

Comment:
Python now provides the higher-level [`pathlib` module](https://docs.python.org/3/library/pathlib.html#pathlib.Path.iterdir), which is more [OOP](https://docs.python.org/3/library/pathlib.html#corresponding-tools):
```suggestion
            for f in pathlib.iterdir("/tmp"):
                if f.isfile() and f.name.startswith(...):

```

Reply:
Changed